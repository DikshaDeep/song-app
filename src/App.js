import React from 'react';
import './App.css';
import { Provider } from "react-redux";
import Routes from "./app/route/index";
import store from "./app/index";

function App() {
  return (
    <Provider store={store}>
      <Routes />
    </Provider>
  );
}

export default App;
