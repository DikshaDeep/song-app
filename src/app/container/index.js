import React from "react";
import styled from 'styled-components'
import {useDispatch, useSelector} from "react-redux";

//  components
import Button from "../component/Button";
import SearchInput from "../component/SearchBox";
import Card from "../component/Card";
//  action
import ActionCreator from "../action/actionCreator";
import bgImage from "../images/maxresdefault.jpg";

const HomeContainer = styled.div`
  padding: 10px;
  text-align: center;
  background-color: transparent;
  background-image: url(${bgImage});
  background-repeat: no-repeat;
  background-size: cover;
`
const SearchContainer = styled.div`
  display: flex;
  margin: 10px 0px;
  width: 100%;
`
const Container = styled.div`
  margin: 0px;
`

function CheckContainer () {
  
  const dispatch = useDispatch();
  const [searchTerm, setTerm] = React.useState("");
  const [player, setPlayer] = React.useState("pause");
  const [audio, setAudio] = React.useState("");
  React.useEffect(() => {
    dispatch(ActionCreator.listRequest("latest"))
  }, [dispatch])

  const data = useSelector(state => state.list);
  const handleSearch = (e) => {
    setTerm(e.target.value)
  }

  const submitSearch = () => {
    dispatch(ActionCreator.listRequest(searchTerm))
  }

  React.useEffect(() => {
    const firstSong = data.list?.results?.find(song => song.kind === "song");
    setAudio(firstSong?.previewUrl)
  }, [data.list])

  const handlePlay = (song, seq) => {
    console.log('song', song, seq)
    if(player === "pause") {
      setPlayer("play")
    } else {
      setPlayer("pause");
      setAudio(song.previewUrl);
    }
  }
  return (
    <HomeContainer>
      <SearchContainer>
        <Container>
          <SearchInput placeholder="Search your song" type="text" value={searchTerm} onChange={(e) => handleSearch(e)} />
        </Container>
        <Container>
          <Button onClick={submitSearch}>Search</Button>
        </Container>
        
      </SearchContainer>
        <Container>
          {data.error?
          <p>{data.errorMessage}</p> :
          <div>
          {data.parseListLoaded ? data.list.results.map((result, i) => (
            <Card 
            key={i}
            seq={i+1}
            url={result.artworkUrl100}
            name={result.collectionName}
            artistName={result.artistName}
            handlePlay={handlePlay}
            player={player}
            result={result}
            audio={audio}
          />)) : <div>Loading....</div>}
          </div>}
        </Container>
    </HomeContainer>
  )
}
export default CheckContainer;
