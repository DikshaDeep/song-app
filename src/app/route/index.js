import React from "react";
import {
  BrowserRouter as Router,
  Switch,
  Route
} from "react-router-dom";
//  components
import PageNotFound from '../container/PageNotFound';
import Home from '../container/index';

export default function appRoute() {
  return (
    <Router>
        <Switch>
          <Route exact path="/">
            <Home />
          </Route>
          <Route exact path="/search">
            <Users />
          </Route>
          <Route>
            <PageNotFound />
          </Route>
        </Switch>
    </Router>
  );
}

function Users() {
  return <h2>Users</h2>;
}