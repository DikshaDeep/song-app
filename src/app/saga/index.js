import { all, fork } from "redux-saga/effects";

//  saga
import list from "./list";

// Redux Saga: Root Saga
export function* rootSaga () {
  yield all([
    fork(list)
  ]);
};
