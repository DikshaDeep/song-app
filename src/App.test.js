import React from 'react';
import { shallow } from './app/enzyme';
import App from './App';

describe('renders App component', () => {
  it('render app', () =>{
    const component = shallow(<App/>);
    expect(component.exists()).toBe(true);
  })
});
