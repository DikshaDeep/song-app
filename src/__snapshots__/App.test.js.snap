// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App component render app 1`] = `ShallowWrapper {}`;
